lib/core/proxy.c

```diff
@@ -622,15 +567,18 @@ static h2o_httpclient_head_cb on_connect(h2o_httpclient_t *client, const char *e
     if (reprocess_if_too_early)
         req->reprocess_if_too_early = 1;
 
-    *body = h2o_iovec_init(NULL, 0);
-    *proceed_req_cb = NULL;
-    if (self->src_req->entity.base != NULL) {
-        *body = self->src_req->entity;
-        if (self->src_req->proceed_req != NULL) {
-            *proceed_req_cb = proceed_request;
-            self->src_req->write_req.cb = write_req;
-            self->src_req->write_req.ctx = self;
-        }
+    if (self->src_req->entity.base == NULL) {
+        body->type = H2O_HTTPCLIENT_REQ_BODY_NONE;
```

Comment:
How about clarifying in the doc-comment of `h2o_httpclient_req_body_t` that `type` is initialized to NONE when the on_connect callback is being called? Then, we do not need to set the type to none in every code that uses httpclient.

The alternative approach would be to initially set `type` to an invalid value to detect applications failing to set it to the correct value.